src/dotenv.net/Parser.cs

```diff
@@ -1,71 +1,89 @@
 using System;
 using System.Collections.Generic;
+using System.Text;
 
 namespace dotenv.net;
 
 internal static class Parser
 {
-    private static readonly char[] SingleQuote = { '\'' };
-    private static readonly char[] DoubleQuotes = { '"' };
+    private const string SingleQuote = "'";
+    private const string DoubleQuotes = "\"";
 
-    internal static ReadOnlySpan<KeyValuePair<string, string>> Parse(ReadOnlySpan<string> dotEnvRows,
-        bool shouldTrimValue)
+    internal static ReadOnlySpan<KeyValuePair<string, string>> Parse(ReadOnlySpan<string> rawEnvRows,
+        bool trimValues)
     {
-        var validEntries = new List<KeyValuePair<string, string>>();
+        var keyValuePairs = new List<KeyValuePair<string, string>>();
 
-        foreach (var dotEnvRow in dotEnvRows)
+        for (var i = 0; i < rawEnvRows.Length; i++)
         {
-            var row = new ReadOnlySpan<char>(dotEnvRow.TrimStart().ToCharArray());
-
-            if (row.IsEmpty)
-                continue;
-
-            if (row.IsComment())
-                continue;
-
-            if (row.HasNoKey(out var index))
-                continue;
-
-            var key = row.Key(index);
-            var value = row.Value(index, shouldTrimValue);
-            validEntries.Add(new KeyValuePair<string, string>(key, value));
+            var rawEnvRow = rawEnvRows[i];
+
+            if(rawEnvRow.StartsWith("#")) continue;
+
+            if (rawEnvRow.Contains("=\""))
```

Comment:
This line will fail if the env is 

```
DOUBLE_QUOTE_MULTI_LINE= "dou
bler"
```

I have merged the PR in but will  have to introduce a fix